supereeg/model.py

```diff
@@ -644,110 +653,6 @@ def _force_update(mo, bo, width=20):
-#
-#     # grab permuted locations
-#     perm_locs = self.locs.iloc[perm_inds]
-#
-#     return model_corrmat_x, loc_label, perm_locs
-#
-# def _some_overlap(self, bo, model_corrmat_x, joint_model_inds, width=20):
-#     """ Compute model when there is some overlap """
-#
-#     # get subject indices where subject locs do not overlap with model locs
-#
-#     bool_bo_mask= np.sum([(bo.get_locs() == y).all(1) for idy, y in self.locs.iterrows()], 0).astype(bool)
-#     disjoint_bo_inds = np.where(~bool_bo_mask)[0]
-#     # d = cdist(bo.get_locs(), self.locs)
-#     # disjoint_bo_inds = np.where(np.isclose(d, 0))[0]
-#
-#     # permute the correlation matrix so that the inds to reconstruct are on the right edge of the matrix
-#     perm_inds = sorted(set(range(self.locs.shape[0])) - set(joint_model_inds)) + sorted(set(joint_model_inds))
-#     model_permuted = model_corrmat_x[:, perm_inds][perm_inds, :]
-#
-#     # permute the model locations (important for the _rbf calculation later)
-#     model_locs_permuted = self.locs.iloc[perm_inds]
-#
-#     # permute the subject locations arranging them
-#     bo_perm_inds = sorted(set(range(bo.get_locs().shape[0])) - set(disjoint_bo_inds)) + sorted(set(disjoint_bo_inds))
-#     sub_bo = bo.get_locs().iloc[disjoint_bo_inds]
-#
-#     #FIXME: won't this change the brain object that the user passes in?  seems problematic...do we need to copy it first?
-#     bo = copy.deepcopy(bo) #added this line re: FIXME statement...
-#     bo.locs = bo.locs.iloc[bo_perm_inds]
-#     bo.data = bo.data[bo_perm_inds]
-#     bo.kurtosis = bo.kurtosis[bo_perm_inds]
-#
-#     # permuted indices for unknown model locations
-#     perm_inds_unknown = sorted(set(range(self.locs.shape[0])) - set(joint_model_inds))
-#     # expanded _rbf weights
-#     #model__rbf_weights = _rbf(pd.concat([model_locs_permuted, bo.locs]), model_locs_permuted)
-#     model__rbf_weights = _log_rbf(pd.concat([model_locs_permuted, sub_bo]), model_locs_permuted, width=width)
-#
-#     # get model expanded correlation matrix
-#     num_corrmat_x, denom_corrmat_x = _blur_corrmat(model_permuted, model__rbf_weights)
-#
-#     # divide the numerator and denominator
-#     #with np.errstate(invalid='ignore'):
-#     model_corrmat_x = _recover_model(num_corrmat_x, denom_corrmat_x, z_transform=True)
-#
-#     # add back the permuted correlation matrix for complete subject prediction
-#     model_corrmat_x[:model_permuted.shape[0], :model_permuted.shape[0]] = model_permuted
-#
-#     # label locations as reconstructed or observed
-#     loc_label = ['reconstructed'] * len(self.locs.iloc[perm_inds_unknown]) + ['observed'] * len(bo.get_locs())
-#
-#     ## unclear if this will return too many locations
-#     perm_locs = self.locs.iloc[perm_inds_unknown].append(bo.get_locs())
-#
-#     return model_corrmat_x, loc_label, perm_locs
-
 def _recover_model(num, denom, z_transform=False):
     warnings.simplefilter('ignore')
 
```

Comment:
This last division when the model is recovered should probably be done in log space... @lucywowen can you add an issue (I don't think it's a simple fix)

Reply:
issue made